source/framework/core/inc/TRestDataSetPlot.h

```diff
@@ -0,0 +1,190 @@
+/*************************************************************************
+ * This file is part of the REST software framework.                     *
+ *                                                                       *
+ * Copyright (C) 2016 GIFNA/TREX (University of Zaragoza)                *
+ * For more information see https://gifna.unizar.es/trex                 *
+ *                                                                       *
+ * REST is free software: you can redistribute it and/or modify          *
+ * it under the terms of the GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3 of the License, or     *
+ * (at your option) any later version.                                   *
+ *                                                                       *
+ * REST is distributed in the hope that it will be useful,               *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          *
+ * GNU General Public License for more details.                          *
+ *                                                                       *
+ * You should have a copy of the GNU General Public License along with   *
+ * REST in $REST_PATH/LICENSE.                                           *
+ * If not, see https://www.gnu.org/licenses/.                            *
+ * For the list of contributors see $REST_PATH/CREDITS.                  *
+ *************************************************************************/
+
+#ifndef REST_TRestDataSetPlot
+#define REST_TRestDataSetPlot
+
+#include "THStack.h"
+#include "TLatex.h"
+#include "TLegend.h"
+#include "TRestDataSet.h"
+#include "TRestMetadata.h"
+
+/// Perform the plot over datasets
+class TRestDataSetPlot : public TRestMetadata {
+    /// Nested classes for internal use only
+   public:
+    /// Auxiliary class for histograms
+    class HistoInfo {
```

Comment:
```suggestion
    struct HistoInfo {
```

A struct is a class with all public members